### ISSUE TITLE
Const declarations are detected as magic numbers when not inside a function

### ISSUE THREAD

**Issue by GuillaumeBouteille** [SonarOpenCommunity/sonar-cxx#296]
Hi !

It seems that declarations of const variables are detected as magic numbers (cxx:MagicNumber rule) when their are defined outside a function. 

Here are some examples of violations found :

``` c++
const int a = 10; // flagged as magic number

namespace TestNamespace {
        const int b = 20; // flagged as magic number
}

class TestClass {
        static const int c = 30; // flagged as magic number
};

int main() {
        const int d = 40; // ok
        return 0;
}
```

I'm using Sonar v4.5 and the sonar-cxx plugin v0.9.1 .

I tried to go through the code to see where the problem could come from, with no luck.

Regards, 

Guillaume
